### ISSUE TITLE
Example code for `clippy::await_holding_invalid_type` doesn't reproduce

### ISSUE THREAD

**Issue by fenollp** [rust-lang/rust-clippy#13517]
### Summary

I am not able to use `clippy::await_holding_invalid_type` (in a multi crate workspace, with types defined in one crate and used in another).

I minimized the reproducer to just the example given in current documentation: https://rust-lang.github.io/rust-clippy/master/index.html#/await_holding_invalid_type

Here's the repro repo with clippy config: https://github.com/fenollp/repro-clippy-await_holding_invalid_type

The call I use: 
```shell
\cargo clippy --locked --frozen --offline --all-targets --all-features -- --no-deps -A clippy::all -W clippy::await_holding_invalid_type
```

Please tell me how I am doing something wrong + if this is supposed to also work for multicrate workspaces? Thanks!

### Lint Name

clippy::await_holding_invalid_type

### Reproducer

I tried this code:

```rust
pub struct CustomLockType;
pub struct OtherCustomLockType;
pub async fn foo() {
    let _x = CustomLockType;
    let _y = OtherCustomLockType;
    baz().await; // Lint violation
}

async fn baz() -> u32 {
    42
}
```

I expected to see this happen:
* a lint error about `_x` and `_y` being in scope while `await`ing `baz()`.

Instead, this happened:
* no errors show up at all

### Version

rustc 1.81.0 (eeb90cda1 2024-09-04)
binary: rustc
commit-hash: eeb90cda1969383f56a2637cbd3037bdf598841c
commit-date: 2024-09-04
host: x86_64-unknown-linux-gnu
release: 1.81.0
LLVM version: 18.1.7

**Comment by fenollp** [comment]
Thanks! This helped me make it work. Indeed I had to tune the type path.
